jdaviz/app.py

```diff
@@ -1162,11 +1162,14 @@ def _update_selected_data_items(self, viewer_id, selected_items):
 
         # Sets the plot axes labels to be the units of the most recently
         # active data.
-        if len(active_data_labels) > 0:
+        n_active = len(active_data_labels)
+        if n_active > 0:
             active_data = self.data_collection[active_data_labels[0]]
             if (hasattr(active_data, "_preferred_translation")
                     and active_data._preferred_translation is not None):
                 self._set_plot_axes_labels(viewer_id)
+            if n_active == 1:
```

Comment:
This does fix the case where all but one entry are unchecked in the data menu, but I can still reproduce the bug if there were three original data layers and then removing whichever is the currently "visible through blinking" (leaving two checked, but hidden by blinking).

Reply:
Ah, in that case, this is not the right fix. I need to ponder more. Thanks for pointing that out!